frontend/app/blueprints/groups/templates/shared.html

```diff
@@ -0,0 +1,119 @@
+                placeholder="Number of samples" required>
+        <input type="text" name="similar-samples-threshold" 
+                id="similar-samples-threshold" class="form-control form-control-sm mt-2" 
+                placeholder="Min similarity" value="0.95" required>
+        <button type="button" id="similar-samples-button" class="btn btn-sm btn-outline-success mt-2"
+                onclick="getSimilarSamplesAndCheckRows(this)">
+            <i class="bi bi-search"></i>
+            Search
+        </button>
+    </form>
+</div>
+{% endmacro %}
+
+{% macro search_similar_js() %}
+async function getSimilarSamples() {
+    const sampleId=window.getSelectedRows()[0]
+    const limit=document.getElementById("similar-samples-limit").value
+    const similarity=document.getElementById("similar-samples-threshold").value
+    const baseUrl = {{ request.script_root|tojson }}
+    try {
+        const response = await fetch(`${baseUrl}/samples/${sampleId}/similar`, {
+            method: "POST",
+            body: JSON.stringify({limit: limit, similarity: similarity}),
+            headers: {
+                'Accept': 'application/json', 
+                'Content-Type': 'application/json' 
+            },
+            credentials: 'same-origin'
+        })
+        const jsonData = await response.json()
+        // check return code
+        if ( !response.ok ) {
+            console.log(`Error while finding similar samples: ${jsonData.details}`)
+            return false
+        } else {
+            return jsonData
+        }
+    } catch ( error ) {
+        console.error("Error:", error)
+        return false
+    }
+}
+
+async function getSimilarSamplesAndCheckRows(btn) {
+    // get similar samples and check rows
+    const parentElem = btn.parentElement.parentElement
+    showSpinner(parentElem)
+    // start query
+    getSimilarSamples().then( async (job) => {
+        if ( job ) {
+            //async () => await fetchJobStatus(job.id),  // function to get job status
+            try {
+                let result = await poll(
+                    async () => fetchJobStatus(job.id),  // function to get job status
+                    validateJobStatus,  // validator
+                    resultParser,       // parse results
+                    3000                // interval time
+                )
+                window.action('select', result.map(sample => sample.sample_id))
+                hideSpinner(parentElem)
+            } catch (err) {
+                throwSmallToast("Error while finding similar samples")
+                hideSpinner(parentElem)
+            }
+        } else {
+            throwSmallToast("Error while finding similar samples")
+            hideSpinner(parentElem)
+        }
+    })
+}
+
+document.getElementById("similar-samples-threshold").addEventListener("input", (elem) => {
+    document.getElementById("similar-samples-button").disabled = isNaN(parseInt(elem.data))
```

Comment:
I think this would be easier to follow:

```
document.getElementById("similar-samples-threshold").addEventListener("input", (threshold_elem) => {
 const threshold_value = threshold_elem.data
 document.getElementById("similar-samples-button").disabled = isNaN(parseInt(threshold_value))
});
```